### PR TITLE
Add date time parse test

### PR DESCRIPTION
**Pull request by tarekgh**

### CODE REVIEW

**Comment by tarekgh** [comment]
@stephentoub I did rebase but still seeing a couple of commits showing in my PR. I don't really know why this is happening. but the file changes is just my change only.

File: src/System.Runtime/tests/System/DateTime.cs [hunk 1]
```diff
@@ -327,6 +327,54 @@ public static void TestGetDateTimeFormats()
...
     [Fact]
+    public static void TestDateTimeParsingWithSpecialCultures()
+    {
+        // Test DateTime parsing with cultures which has the date separator and time separator are same
+        string[] cultureNames = new string[]
+        {
+            "fi-FI",
+            "nb-NO",
+            "nb-SJ"
+        };
```
Comment (Maxwe11) [inline src/System.Runtime/tests/System/DateTime.cs]: Why not to use `[Theory]`?

Comment (tarekgh) [inline src/System.Runtime/tests/System/DateTime.cs]: yes this can be a good idea too.

**Comment by tarekgh** [comment]
@ellismg could you please have a quick look?

**Comment by ellismg** [comment]
Code changes look fine, but I think we need to address the history issue.  Can you try the following?

```
> git checkout AddDateTimeParseTest
> git fetch upstream
> git reset --hard upstream/master
> git cherry-pick cd48895
> git cherry-pick 46d1d74
> git push origin +AddDateTimeParseTest
```

This will take your local copy off the AddDateTimeParseTest branch and reset it back so it matches what's in corefx/master.  Then the cherry-pick command will take your two changes and replay them on top of this new branch.

Finally, we push the new history to GitHub.

**Comment by tarekgh** [comment]
@ellismg thanks Matt, looks your steps took care with the history issue.

**Pull request merged**